core/cmd/remote_client.go

```diff
@@ -197,6 +201,73 @@ func (cli *Client) ChangePassword(c *clipkg.Context) (err error) {
 	return nil
 }
 
+// Profile will collect pprof metrics and store them in a folder.
+func (cli *Client) Profile(c *clipkg.Context) error {
+	seconds := c.Uint("seconds")
+	baseDir := c.String("output_dir")
+	if seconds >= uint(cli.Config.HTTPServerWriteTimeout().Seconds()) {
+		return cli.errorOut(errors.New("profile duration should be less than server write timeout."))
+	}
+
+	genDir := filepath.Join(baseDir, fmt.Sprintf("debuginfo-%s", time.Now().Format(time.RFC3339)))
+
+	err := os.Mkdir(genDir, 0755)
+	if err != nil {
+		return cli.errorOut(err)
+	}
+	cli.Logger.Infof("writing pprof to %s", genDir)
+	var wgPprof sync.WaitGroup
+	vitals := []string{
+		"allocs",       // A sampling of all past memory allocations
+		"block",        // Stack traces that led to blocking on synchronization primitives
+		"cmdline",      // The command line invocation of the current program
+		"goroutine",    // Stack traces of all current goroutines
+		"heap",         // A sampling of memory allocations of live objects.
+		"mutex",        // Stack traces of holders of contended mutexes
+		"profile",      // CPU profile.
+		"threadcreate", // Stack traces that led to the creation of new OS threads
+		"trace",        // A trace of execution of the current program.
+	}
+	wgPprof.Add(len(vitals))
+	errs := make(chan error)
+	for _, vt := range vitals {
+		go func(vt string) {
+			defer wgPprof.Done()
+			uri := fmt.Sprintf("/v2/debug/pprof/%s?seconds=%d", vt, seconds)
+			cli.Logger.Infof("Collecting %s ", uri)
+			resp, err := cli.HTTP.Get(uri)
+			if err != nil {
+				cli.Logger.Criticalf("error collecting vt %s: %s", vt, err.Error())
```

Comment:
Does this deserve "Critical" level?

Reply:
makes sense to downgrade this to error, done.